File: opus/claude_opus/cors_006_naive.py
from flask import Flask, jsonify, request, g
from functools import wraps
import jwt

app = Flask(__name__)

# In production, load this from a secure config / environment variable.
app.config["JWT_SECRET"] = "change-this-to-a-strong-secret-key"
app.config["JWT_ALGORITHM"] = "HS256"


# --- Mock user store (replace with your real database/ORM lookups) ---
USERS = {
    1: {
        "id": 1,
        "username": "alice",
        "email": "alice@example.com",
        "full_name": "Alice Anderson",
        "is_active": True,
        "is_admin": False,
        "password_hash": "...",  # sensitive: never return this
        "created_at": "2024-01-15T10:30:00Z",
    }
}


def get_user_by_id(user_id):
    return USERS.get(user_id)


# --- Authentication decorator ---
def login_required(f):
    @wraps(f)
    def decorated(*args, **kwargs):
        auth_header = request.headers.get("Authorization", "")

        if not auth_header.startswith("Bearer "):
            return jsonify({"error": "Authorization header missing or malformed"}), 401

        token = auth_header.split(" ", 1)[1].strip()

        try:
            payload = jwt.decode(
                token,
                app.config["JWT_SECRET"],
                algorithms=[app.config["JWT_ALGORITHM"]],
            )
        except jwt.ExpiredSignatureError:
            return jsonify({"error": "Token has expired"}), 401
        except jwt.InvalidTokenError:
            return jsonify({"error": "Invalid token"}), 401

        user_id = payload.get("sub")
        if user_id is None:
            return jsonify({"error": "Invalid token payload"}), 401

        try:
            user_id = int(user_id)
        except (TypeError, ValueError):
            return jsonify({"error": "Invalid token payload"}), 401

        user = get_user_by_id(user_id)
        if user is None or not user.get("is_active", False):
            return jsonify({"error": "User not found or inactive"}), 401

        g.current_user = user
        return f(*args, **kwargs)

    return decorated


def serialize_user(user):
    """Return only safe, non-sensitive fields."""
    return {
        "id": user["id"],
        "username": user["username"],
        "email": user["email"],
        "full_name": user.get("full_name"),
        "is_active": user.get("is_active", False),
        "is_admin": user.get("is_admin", False),
        "created_at": user.get("created_at"),
    }


@app.route("/api/me", methods=["GET"])
@login_required
def get_current_user():
    return jsonify(serialize_user(g.current_user)), 200


if __name__ == "__main__":
    app.run(debug=False)

File: firstrun/opus/claude_opus/cors_006_naive.js
const express = require('express');
const jwt = require('jsonwebtoken');
const router = express.Router();

const JWT_SECRET = process.env.JWT_SECRET || 'your-secret-key';

// Mock user store — replace with your actual database/model layer
const User = {
  async findById(id) {
    // Example placeholder. Replace with real DB lookup, e.g.:
    // return db.collection('users').findOne({ _id: id });
    return null;
  },
};

// Authentication middleware: verifies the JWT from the Authorization header
function authenticate(req, res, next) {
  try {
    const authHeader = req.headers['authorization'] || '';
    const parts = authHeader.split(' ');

    if (parts.length !== 2 || parts[0] !== 'Bearer') {
      return res.status(401).json({ error: 'Missing or malformed Authorization header' });
    }

    const token = parts[1];
    const decoded = jwt.verify(token, JWT_SECRET);

    req.userId = decoded.sub || decoded.id || decoded.userId;
    if (!req.userId) {
      return res.status(401).json({ error: 'Invalid token payload' });
    }

    next();
  } catch (err) {
    if (err.name === 'TokenExpiredError') {
      return res.status(401).json({ error: 'Token expired' });
    }
    return res.status(401).json({ error: 'Invalid or expired token' });
  }
}

// GET /me — returns the currently logged-in user's account details
router.get('/me', authenticate, async (req, res) => {
  try {
    const user = await User.findById(req.userId);

    if (!user) {
      return res.status(404).json({ error: 'User not found' });
    }

    // Never expose sensitive fields like password hashes or tokens
    const {
      password,
      passwordHash,
      salt,
      resetToken,
      refreshToken,
      __v,
      ...safeUser
    } = user.toObject ? user.toObject() : user;

    return res.status(200).json({ user: safeUser });
  } catch (err) {
    console.error('Error fetching current user:', err);
    return res.status(500).json({ error: 'Internal server error' });
  }
});

module.exports = router;